src/decision/src/LidarDecision.cpp

```diff
@@ -131,11 +131,27 @@ LidarDecision::LidarDecision(int argc, char **argv, std::string node_name) {
         twist.angular.y = 0;
         twist.angular.z = 0;
 
-        if (obstacle.getAvgDistance() < danger_distance && abs(obstacle.getAvgAngle()) < danger_angle) {
-            // TODO: Improve the algorithm here to at least use some sort of scale  (linear or otherwise) dependent on obstacle distance and angle
-            // Check if we should be turning left or right
-            int left_or_right = (obstacle.getAvgAngle() > 0) ? -1 : 1;
-            twist.angular.z = angular_vel_multiplier * left_or_right;
+        if (obstacle.getMinDistance() <= danger_distance && abs(obstacle.getAvgAngle()) <= danger_angle) {
+            float minDistance = obstacle.getMinDistance();
+            float distance_score_threshold = 1.0;
```

Comment:
You're only using this once, so I don't think that we really need to make it a separate variable. Don't think it really helps to make anything clearer.